lib/client/retryInterceptor.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * Sleeps for given amount of seconds
+ *
+ * @param seconds
+ * @returns {Promise}
+ */
+function sleep(seconds) {
+    return new Promise(resolve => setTimeout(() => resolve(), seconds * 1000));
+}
+
+/**
+ * Attach the retry interceptor to the axios instance to transparently ensure all requests are going through eventually.
+ *
+ * @param {Object} customAxios axios instance
+ * @return {Object}
+ */
+module.exports = (customAxios) => {
+    // on 429 Too Many Requests responses retry after 5 seconds
+    customAxios.interceptors.response.use(response => response,
+        async (error) => {
+            if (error.response && error.response.status === 429) {
+                await sleep(5);
+
+                return customAxios(error.config);
+            }
+
+            return Promise.reject(error);
+        });
```

Comment:
Please move the closing `)` to the next line.

Reply:
should i move `response => response,` to the next line as well for consistency?

Reply:
Yea, good point.